projects/components/src/table/table.component.ts

```diff
@@ -648,6 +657,53 @@ export class TableComponent
     this.changeDetector.detectChanges();
   }
 
+  private downloadCsv(): void {
```

Comment:
This file is a crime against humanity already, would refactor as much of this complexity outside as possible

Reply:
I had mainly meant to reduce file size, but now that you mention it would be at least worth considering whether table controls would be a better home to this (not sure haven't looked in there in a while and not sure what it has access to, but this _is_ a control).

Reply:
Table controls do not have access to data. Only thing it can do is to be a proxy between the table and the messaging service we added. Since we have a messaging service, I think we should avoid another intermediary.

Reply:
Messaging service? can you point me to that? I looked at the controls and agree they don't have access to data, but I always see them being wired directly to table (which agree, isn't a great model - should have accept the table as content and wired directly to it).

Reply:
@aaron-steinfeld I have added a `TableCsvDownloaderService` which is just a way to let the table know that it should download CSV.